staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go

```diff
@@ -189,15 +190,28 @@ func ValidateObjectMetaAccessor(meta metav1.Object, requiresNamespace bool, name
 	return allErrs
 }
 
+// ValidateInitializerName validates that the name is fully qualified
+func ValidateInitializerName(fldPath *field.Path, name string) field.ErrorList {
+	var allErrors field.ErrorList
+	if len(name) == 0 {
+		allErrors = append(allErrors, field.Required(fldPath, ""))
+	}
+	if errs := validationutil.IsDNS1123Subdomain(name); len(errs) > 0 {
+		allErrors = append(allErrors, field.Invalid(fldPath, name, strings.Join(errs, ",")))
+	}
+	if len(strings.Split(name, ".")) < 3 {
+		allErrors = append(allErrors, field.Invalid(fldPath, name, "should be a domain with at least two dots"))
+	}
+	return allErrors
+}
+
 func ValidateInitializers(initializers *metav1.Initializers, fldPath *field.Path) field.ErrorList {
 	var allErrs field.ErrorList
 	if initializers == nil {
 		return allErrs
 	}
 	for i, initializer := range initializers.Pending {
-		for _, msg := range validation.IsQualifiedName(initializer.Name) {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("pending").Index(i), initializer.Name, msg))
-		}
+		allErrs = append(allErrs, ValidateInitializerName(fldPath.Child("pending").Index(i).Child("name"), initializer.Name)...)
```

Comment:
Was it possible for me to submit an object that contained a pending initializer before (not rely on admission to set it)?  If so, an action-required release note is probably in order, since tightening validation may accidentally block deletion (finalizers), right?

Reply:
It is an alpha feature, but yes, worth at least reminding people that alpha features can wedge your cluster.